R/license.R

```diff
@@ -4,150 +4,190 @@
+  version <- check_license_version(version, c(2.1, 3))
+  if (is_package()) {
+    abbr <- license_abbr("LGPL", version, include_future)
+    use_description_field("License", abbr, overwrite = TRUE)
+  }
+  use_license_template(glue("LGPL-{version}"))
 }
 
 #' @rdname licenses
 #' @export
-use_apl2_license <- function(name = find_name()) {
-  force(name)
-  check_is_package("use_apl2_license()")
+use_apl_license <- function(version = 2, include_future = TRUE) {
+  version <- check_license_version(version, 2)
 
-  use_description_field("License", "Apache License (>= 2.0)", overwrite = TRUE)
-  use_license_template("apache-2.0", name)
+  if (is_package()) {
+    abbr <- license_abbr("Apache License", version, include_future)
+    use_description_field("License", abbr, overwrite = TRUE)
+  }
+  use_license_template(glue("apache-{version}"))
+}
+
+#' @rdname licenses
+#' @export
+use_cc0_license <- function() {
+  if (is_package()) {
+    use_description_field("License", "CC0", overwrite = TRUE)
+  }
+  use_license_template("cc0")
 }
 
 #' @rdname licenses
 #' @export
-use_cc0_license <- function(name = find_name()) {
-  force(name)
-  check_is_package("use_cc0_license()")
+use_ccby_license <- function() {
+  if (is_package()) {
+    use_description_field("License", "CC BY 4.0", overwrite = TRUE)
+  }
+  use_license_template("ccby-4")
+}
+
+
+# Fallbacks ---------------------------------------------------------------
 
-  use_description_field("License", "CC0", overwrite = TRUE)
-  use_license_template("cc0", name)
+#' @rdname licenses
+#' @export
+#' @usage NULL
+use_gpl3_license <- function() {
+  use_gpl_license(3)
 }
 
 #' @rdname licenses
 #' @export
-use_ccby_license <- function(name = find_name()) {
-  force(name)
-  check_is_package("use_ccby_license()")
+#' @usage NULL
+use_agpl3_license <- function() {
+  use_agpl_license(3)
+}
 
-  use_description_field("License", "CC BY 4.0", overwrite = TRUE)
-  use_license_template("ccby-4", name)
+#' @rdname licenses
+#' @export
+#' @usage NULL
+use_apl2_license <- function() {
+  use_apl_license(2)
 }
 
-use_license_template <- function(license, name) {
+# Helpers -----------------------------------------------------------------
+
+use_license_template <- function(license, data = list()) {
   license_template <- glue("license-{license}.md")
 
-  use_template(
-    license_template,
+  use_template(license_template,
     save_as = "LICENSE.md",
-    data = license_data(name),
+    data = data,
     ignore = TRUE
   )
 }
 
-license_data <- function(name, base_path = proj_get()) {
-  list(
-    year = format(Sys.Date(), "%Y"),
-    name = name,
-    project = project_name(base_path)
-  )
-}
-
+check_license_version <- function(version, possible) {
+  version <- as.double(version)
 
-find_name <- function() {
-  name <- getOption("usethis.full_name")
-  if (!is.null(name)) {
-    return(name)
+  if (!version %in% possible) {
+    possible <- glue_collapse(possible, sep = "", last = "or, ")
```

Comment:
```suggestion
    possible <- glue_collapse(possible, sep = ", ", last = " or ")
```
Not sure exactly what inputs are anticipated but this is more what I expected to see.

Reply:
Oops!